src/test/java/com/softserve/academy/spaced/repetition/util/DomainFactory.java

```diff
@@ -0,0 +1,255 @@
+        return course;
+    }
+
+
+    public static Folder createFolder(Long folderId, Set<Deck> decks) {
+        Folder folder = new Folder();
+        folder.setId(folderId);
+        folder.setDecks(decks);
+        return folder;
+    }
+
+    public static Deck createDeck(Long deckId, String name, String description, String syntaxToHighLight,
+                                  Category category, double rating, User deckOwner, List<Card> cards,
+                                  List<Course> courses, List<DeckRating> deckRatings, Set<Folder> folders,
+                                  List<DeckComment> deckComments) {
+        Deck deck = new Deck();
+        deck.setId(deckId);
+        deck.setName(name);
+        deck.setDescription(description);
+        deck.setSynthaxToHighlight(syntaxToHighLight);
+        deck.setCategory(category);
+        deck.setRating(rating);
+        deck.setDeckOwner(deckOwner);
+        deck.setCards(cards);
+        deck.setCourses(courses);
+        deck.setDeckRatings(deckRatings);
+        deck.setFolders(folders);
+        deck.setDeckComments(deckComments);
+        return deck;
+    }
+
+    public static DeckComment createDeckComment(Long deckCommentId, String commentText, Date commentDate, Person person,
+                                                Long parentCommentId, Deck deck) {
+        DeckComment deckComment = new DeckComment();
+        deckComment.setId(deckCommentId);
+        deckComment.setCommentText(commentText);
+        deckComment.setCommentDate(commentDate);
+        deckComment.setPerson(person);
+        deckComment.setParentCommentId(parentCommentId);
+        deckComment.setDeck(deck);
+        return deckComment;
+    }
+
+    public static DeckRating createDeckRating(Long deckRatingId, String accountEmail, Deck deck, int rating) {
+        DeckRating deckRating = new DeckRating();
+        deckRating.setId(deckRatingId);
+        deckRating.setAccountEmail(accountEmail);
+        deckRating.setDeck(deck);
+        deckRating.setRating(rating);
+        return deckRating;
+    }
+
+    public static Card createCard(Long id, String title, String question, String answer, Deck deck) {
+        Card card = new Card();
+        card.setId(id);
+        card.setTitle(title);
+        card.setQuestion(question);
+        card.setAnswer(answer);
+        card.setRating(0L);
+        card.setDeck(deck);
+        return card;
+    }
+
+    public static Card createCard(long rating, long deckId) {
+        Deck deck = new Deck();
+        deck.setId(deckId);
+        Card card = new Card();
+        card.setId(1L);
+        card.setTitle("Card 1");
+        card.setAnswer("There are 4 access modifiers in Java: public, protected, default and private");
+        card.setQuestion("How many access modifiers do you know in Java?");
+        card.setDeck(deck);
+        card.setRating(rating);
+        return card;
+    }
+
+    public static List<Card> createLearningCards(long deckId) {
```

Comment:
it's not a domain

Reply:
this method used by 2 tests, so this is the large part of duplicated code

Reply:
Okay, i agree with you, but it's a not domain.
Review your tests that uses this list.

Reply:
Why do you move this method from controller, if it not use in other tests?

Reply:
And why do you touch controller?
We should rewrite-create tests only for services.

Reply:
ok, I`ll fix it